modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/FilteringVectorProcess.java

```diff
@@ -0,0 +1,423 @@
+            namespaceSupport = new NamespaceSupport();
+            for (Iterator it = namespaces.entrySet().iterator(); it.hasNext(); ) {
+                Map.Entry entry = (Map.Entry) it.next();
+                String prefix = (String) entry.getKey();
+                String namespace = (String) entry.getValue();
+                namespaceSupport.declarePrefix(prefix, namespace);
+            }
+        }
+        return namespaceSupport;
+    }
+
+    /**
+     * A FeatureCollection wrapper to filter out features according to the aggregation parameter and
+     * the groups defined by the groupingAttributes
+     */
+    static class GroupingFeatureCollection
+            extends DecoratingFeatureCollection<FeatureType, Feature> {
+
+        List<PropertyName> groupingAttributes;
+
+        PropertyName operationAttribute;
+
+        Operations aggregation;
+
+        public GroupingFeatureCollection(
+                FeatureCollection<FeatureType, Feature> delegate,
+                List<PropertyName> groupingAttributes,
+                PropertyName operationAttribute,
+                Operations aggregation) {
+            super(delegate);
+            this.groupingAttributes = groupingAttributes;
+            this.operationAttribute = operationAttribute;
+            this.aggregation = aggregation;
+        }
+
+        @Override
+        public FeatureIterator<Feature> features() {
+            return new GroupingFeatureIterator(
+                    new PushBackFeatureIterator(delegate.features()),
+                    groupingAttributes,
+                    operationAttribute,
+                    aggregation);
+        }
+    }
+
+    /**
+     * A FeatureIterator wrapper to filter out features according to the aggregation parameter and
+     * the groups defined by the groupingAttributes
+     */
+    static class GroupingFeatureIterator extends DecoratingFeatureIterator<Feature> {
+
+        private List<PropertyName> groupByAttributes;
+
+        private PropertyName operationAttribute;
+
+        private Operations aggregation;
+
+        private Feature next;
+
+        /**
+         * Wrap the provided FeatureIterator.
+         *
+         * @param iterator Iterator to be used as a delegate.
+         */
+        public GroupingFeatureIterator(
+                PushBackFeatureIterator iterator,
+                List<PropertyName> groupByAttributes,
+                PropertyName operationValue,
+                Operations aggregation) {
+            super(iterator);
+            this.groupByAttributes = groupByAttributes;
+            this.operationAttribute = operationValue;
+            this.aggregation = aggregation;
+        }
+
+        @Override
+        public boolean hasNext() {
+            List<Object> groupingValues = new ArrayList<>(groupByAttributes.size());
+            Map<Object, Feature> beingFiltered = new HashMap<>();
+            List<Double> valuesToCompare = new ArrayList<>();
```

Comment:
Why do they need to be doubles? Any object supporting equals would work as a grouping mechanism here.

Reply:
fixed